src/user/user.controller.ts

```diff
@@ -84,7 +83,7 @@ export class UserController {
   @Patch()
   @UseGuards(FirebaseAuthGuard)
   async updateUser(@Body() updateUserDto: UpdateUserDto, @Req() req: Request) {
```

Comment:
It'd be good to add the return type `: Promise<UserEntity>` to the `updateUser` function here and in the controller!